server.lua

```diff
@@ -168,32 +194,73 @@ end)
 
 RegisterNetEvent('PoliceEMSActivity:RegisterUser')
 AddEventHandler('PoliceEMSActivity:RegisterUser', function()
-	local src = source
+	RegisterUser(source)
+end)
+
+-- EXPORT FUNCTIONS
+function IsPlayerOnDuty(player)
+	if onDuty[player] ~= nil then
+		return true
+	else
+		return false
+	end
+end
+
+function GetPlayerBlipTag(player)
+	return activeBlip[player][1]
+end
+-- END EXPORT FUNCTIONS
+
+function RegisterUser(user)
+	local src = tonumber(user)
+
 	for k, v in ipairs(GetPlayerIdentifiers(src)) do
-			if string.sub(v, 1, string.len("discord:")) == "discord:" then
-				identifierDiscord = v
-			end
+		if string.sub(v, 1, string.len("discord:")) == "discord:" then
+			identifierDiscord = v
+		end
 	end
+
 	local perms = {}
+
 	if identifierDiscord then
 		local roleIDs = exports.Badger_Discord_API:GetDiscordRoles(src)
 		if not (roleIDs == false) then
 			for k, v in pairs(roleList) do
 				for j = 1, #roleIDs do
-					if exports.Badger_Discord_API:CheckEqual(v[1], roleIDs[j]) then
-						table.insert(perms, k);
-						activeBlip[src] = k;
-						hasPerms[src] = true;
-						print("[PEA] Gave Perms Sucessfully")
+					if exports.Badger_Discord_API:CheckEqual(k, roleIDs[j]) then
+
+						for _, t in pairs(v) do
+							table.insert(perms, t);
+							activeBlip[src] = t;
+							hasPerms[src] = true;
+							print("[PEA] Gave " .. GetPlayerName(src) .. " Perms Sucessfully")
+						end
+
+						if Config.EnableInheritances then
+							for role, t in pairs(inheritances) do
+								if exports.Badger_Discord_API:CheckEqual(role, roleIDs[j]) then
+									for i = 1, #t do
+										for _, blip in pairs(roleList[t[i]]) do
+											table.insert(perms, blip);
+											hasPerms[src] = true;
+										end
+									end
+									
+									print("[PEA] Gave " .. GetPlayerName(src) .. " Perms Sucessfully via Inheritance.")
+								end
+							end
+						end
 					end
 				end
 			end
+
 			permTracker[src] = perms;
 		else
 			print("[PoliceEMSActivity] " .. GetPlayerName(src) .. " has not gotten their permissions cause roleIDs == false")
 		end
 	else
 		print("[PoliceEMSActivity] " .. GetPlayerName(src) .. " has not gotten their permissions cause discord was not detected...")
 	end
+
 	permTracker[src] = perms; 
-end)
+end
```

Comment:
Doesn't this still need the `)` ?

Reply:
No I moved the register user code to a function.